src/main/java/com/mraof/minestuck/client/ClientProxy.java

```diff
@@ -213,28 +217,50 @@ public boolean addDestroyEffects(BlockState state, Level level, BlockPos pos, Pa
 			}
 		}, MSBlocks.GATE, MSBlocks.GATE_MAIN, MSBlocks.RETURN_NODE, MSBlocks.RETURN_NODE_MAIN);
 		
-		for(DeferredHolder<FluidType, FluidType> fluidType : Arrays.asList(MSFluids.OIL_TYPE, MSFluids.LIGHT_WATER_TYPE,
-				MSFluids.BLOOD_TYPE, MSFluids.BRAIN_JUICE_TYPE, MSFluids.WATER_COLORS_TYPE, MSFluids.ENDER_TYPE,
-				MSFluids.CAULK_TYPE, MSFluids.MOLTEN_AMBER_TYPE))
+		event.registerFluidType(commonFluidExtension(MSFluids.OIL_TYPE.getId(), new Vector3f(0F, 0F, 0F), 0.8F), MSFluids.OIL_TYPE);
+		event.registerFluidType(commonFluidExtension(MSFluids.LIGHT_WATER_TYPE.getId(), new Vector3f(0.2F, 0.3F, 1F), 0.2F), MSFluids.LIGHT_WATER_TYPE);
+		event.registerFluidType(commonFluidExtension(MSFluids.BLOOD_TYPE.getId(), new Vector3f(0.8F, 0F, 0F), 0.35F), MSFluids.BLOOD_TYPE);
+		event.registerFluidType(commonFluidExtension(MSFluids.BRAIN_JUICE_TYPE.getId(), new Vector3f(0.55F, 0.25F, 0.7F), 0.25F), MSFluids.BRAIN_JUICE_TYPE);
+		event.registerFluidType(new IClientFluidTypeExtensions()
 		{
-			event.registerFluidType(new IClientFluidTypeExtensions()
+			private final ResourceLocation stillTexture = MSFluids.WATER_COLORS_TYPE.getId().withPrefix("block/still_");
+			private final ResourceLocation flowingTexture = MSFluids.WATER_COLORS_TYPE.getId().withPrefix("block/flowing_");
+			
+			@Override
+			public ResourceLocation getStillTexture()
 			{
-				private final ResourceLocation stillTexture = fluidType.getId().withPrefix("block/still_");
-				private final ResourceLocation flowingTexture = fluidType.getId().withPrefix("block/flowing_");
-				
-				@Override
-				public ResourceLocation getStillTexture()
-				{
-					return stillTexture;
-				}
+				return stillTexture;
+			}
+			
+			@Override
+			public ResourceLocation getFlowingTexture()
+			{
+				return flowingTexture;
+			}
+			
+			@Override
+			public Vector3f modifyFogColor(Camera camera, float partialTick, ClientLevel level, int renderDistance, float darkenWorldAmount, Vector3f fluidFogColor)
+			{
+				Vector3f colorValue = new Vector3f(0F, 20F, 30F)
+						.rotateY((float) (camera.getEntity().getX() / 2))
+						.rotateX((float) (camera.getEntity().getZ() / 2))
+						.rotateY((float) (camera.getEntity().getY()))
+						.normalize();
 				
-				@Override
-				public ResourceLocation getFlowingTexture()
-				{
-					return flowingTexture;
-				}
-			}, fluidType);
-		}
+				return new Vector3f(colorValue.x % 1F, colorValue.y % 1F, colorValue.z % 1F);
+			}
+			
+			@Override
+			public void modifyFogRender(Camera camera, FogRenderer.FogMode mode, float renderDistance, float partialTick, float nearDistance, float farDistance, FogShape shape)
+			{
+				RenderSystem.setShaderFogStart(-8);
+				RenderSystem.setShaderFogEnd(48F / 0.2F);
+				RenderSystem.setShaderFogShape(FogShape.SPHERE);
+			}
+		}, MSFluids.WATER_COLORS_TYPE);
```

Comment:
Underwater colors work but fog is seemingly not present

Reply:
Ah. good catch! I'll see if I can figure out what's going on with that.

Reply:
Fix pushed.